go/vt/wrangler/reparent.go

```diff
@@ -426,6 +427,10 @@ func (wr *Wrangler) plannedReparentShardLocked(ctx context.Context, ev *events.R
 	event.DispatchUpdate(ev, "promoting slave")
 	rp, err = wr.tmc.PromoteSlaveWhenCaughtUp(ctx, masterElectTabletInfo.Tablet, rp)
 	if err != nil {
+		// if this fails it is not enough to return an error. we should rollback all the changes made by DemoteMaster
+		if err1 := wr.tmc.UndoDemoteMaster(ctx, oldMasterTabletInfo.Tablet); err1 != nil {
+			log.Warningf("Encountered error %v while trying to undo DemoteMaster", err1)
```

Comment:
If this is encountered, I'm not sure where that leaves the state of the shard.

Reply:
if the previous call to DemoteMaster succeeded, then this should also succeed unless something changed in the meantime. if it does fail, then you are back in today's situation where PromoteSlave failed and the cluster is in an inconsistent state.